pagopa/io-app
# feat: [IOCOM-878] SEND payments bottom sheet with new DS

⚠️ This PR depends on #5650 ⚠️

## Short description
This PR adds the payment's bottom sheet for a SEND message, with the new DS.
![Simulator Screenshot - iPhone 15 - 2024-04-03 at 16 42 28](https://github.com/pagopa/io-app/assets/5150343/5e839a86-ac45-4427-a95d-323e5d01f314)

## List of changes proposed in this pull request
- `MessagePaymentBottomSheet` linked to `MessageDetails`
- Better code sharing of the business logic that computes whether or not to show (and include) the bottom sheet

## How to test
Using the io-dev-api-server, generate some SEND messages:
- a cancelled message shall never have the link to show more payments (use a cancelled message with 6+ paid payments)
- a message with not payments shall never have it
- a message with one payment shall never have it
- a message with up to five payments shall never have it
- a message with more than five payments should always have the link and the bottom sheet

Also check that the payment flow works from the bottom sheet and that the payment data displayed in the main view is the same as the one displayed in the bottom sheet (and that it updates accordingly)
